core/src/main/java/com/sequenceiq/cloudbreak/service/upgrade/rds/RdsUpgradeOrchestratorService.java

```diff
@@ -138,21 +144,35 @@ public void prepareUpgradeEmbeddedDatabase(Long stackId) throws CloudbreakOrches
         hostOrchestrator.runOrchestratorState(stateParams);
     }
 
-    public void validateDbBackupSpace(Long stackId) throws CloudbreakOrchestratorException {
-        Double rootVolumeFreeSpaceMb = getRootVolumeFreeSpace(stackId);
-        Double databaseSizeMb = determineDatabaseSize(stackId);
+    public void determineDbBackupLocation(Long stackId) throws CloudbreakOrchestratorException {
+        StackDto stackDto = stackDtoService.getById(stackId);
+        cleanUpFormerBackups(stackDto);
+        String largestFreeSpacedVolume = getLargestFreeSpacedVolume(stackDto);
+        validateDbBackupSpace(stackDto, largestFreeSpacedVolume);
+        updateRdsUpgradePillar(stackDto, largestFreeSpacedVolume);
+    }
+
+    private void updateRdsUpgradePillar(StackDto stackDto, String rdsBackupLocation) throws CloudbreakOrchestratorException {
+        OrchestratorStateParams stateParams = createStateParams(stackDto, GET_EXTERNAL_DB_SIZE, true);
+        SaltConfig saltConfig = new SaltConfig(upgradeRdsBackupRestoreStateParamsProvider.createParamsForRdsBackupRestore(stackDto, rdsBackupLocation));
+        hostOrchestrator.saveCustomPillars(saltConfig, new ClusterDeletionBasedExitCriteriaModel(stackDto.getId(), stackDto.getCluster().getId()), stateParams);
+    }
+
+    private void validateDbBackupSpace(StackDto stackDto, String backupLocation) throws CloudbreakOrchestratorException {
+        Double freeSpaceMb = getVolumeFreeSpace(stackDto, backupLocation);
```

Comment:
this should not be run again as we have the data from the previous step

Reply:
It would be a more complex logic on java side as there is a fallback logic to `/var` in `getVolumeWithLargestFreeSpace()` method. In this case there is no free space information and we have to call the freespace information command again. So I decided to do it separately, as it is more readable for me. Is this acceptable for you?